### ISSUE TITLE
[OpenMP] Semantic checks for taskloop

### ISSUE THREAD

**Issue by kiranchandramohan** [flang-compiler/f18#570]
Section 2.9.2 of the OpenMP standard describes the Taskloop construct. I am planning to implement the checks. 

Note: This issue is being raised to avoid duplication of work. Please update the projects page.

**Comment by sscalpone** [comment]
Hi Kiran, what are you planning to do?  For example, right now, there are no symbols or scopes for OpenMP as far as I know.  Are you going to sync with Brian & work on legality of clauses in taskloop?

**Comment by kiranchandramohan** [comment]
Yes, I am going to work on the legality of taskloop clauses. Most of the checks seem straightforward and are similar to existing checks, either permitted clauses or clauses that can only be used once. Then a few about two clauses not occurring together, being a positive integer etc. 

I was thinking that these strucutral checks for OpenMP constructs were put in the Projects page so that others can help with the implementation of these checks. I will sync with Brian. @ichoyjx please let me know in case there is something missing before task loop checks can be taken up or if all these checks are something you were planning to do yourself.

**Comment by ichoyjx** [comment]
Hi Kiran, if you are targeting on the checks for the following you mentioned:
> either permitted clauses or clauses that can only be used once. Then a few about two clauses not occurring together, being a positive integer, etc.

Then nothing is missing that may block the implementation. Just FYI, the Projects page listed almost all the TODO directives.

**Comment by kiranchandramohan** [comment]
https://github.com/DavidTruby is working on this.

**Comment by DavidTruby** [comment]
Taskloop checks are now implemented except for branching in and out of a loop, which needs also doing for standard loops.

**Comment by sscalpone** [comment]
@kiranchandramohan You can close this issue if you are satisfied.

**Comment by kiranchandramohan** [comment]
@sscalpone we are waiting for @ichoyjx to implement the branching in and out of a loop. If he will implement for taskloops also then we can close the ticket. Otherwise, this ticket has to remain open as a reminder to use Brian's branch checks for taskloop also.
